### PR TITLE
List plugin options

### PR DESCRIPTION
**Pull request by andrewvc**
Add new `:list `property to configuration parameters.

If set to try this will allow the user to specify one or more values.
This generally replaces the :array type, which had fewer type checks.

The array type is still needed for lists of hashes and mixed types.

### CODE REVIEW

File: logstash-core/lib/logstash/config/mixin.rb [hunk 1]
```diff
@@ -328,58 +328,84 @@ def validate_check_invalid_parameter_names(params)
...
+      
+      return [is_valid, processed_value]
+    end
+
     def validate_check_parameter_values(params)
       # Filter out parametrs that match regexp keys.
       # These are defined in plugins like this:
       #   config /foo.*/ => ... 
-      is_valid = true
+      all_params_valid = true
```
Comment (andrewvc) [inline logstash-core/lib/logstash/config/mixin.rb]: Renamed this and `success` because it was confusing to have two notions of validity with similar names in the same function.

File: docs/static/configuration.asciidoc [hunk 2]
```diff
@@ -83,25 +83,38 @@ The settings you can configure vary according to the plugin type. For informatio
...
 
 [source,js]
 ----------------------------------
   path => [ "/var/log/messages", "/var/log/*.log" ]
-  path => "/data/mysql/mysql.log"
+  uris => [ "http://elastic.co", "http://example.net" ]
 ----------------------------------
 
-This example configures `path` to be an array that contains an element for each of the three strings.
+This example configures `path`, which is a `string` to be a list that contains an element for each of the three strings. It also will configure the `uris` parameter to be a list of URIs, failing if any of the URIs provided are not valid.
```
Comment (jordansissel) [inline docs/static/configuration.asciidoc]: Not part of the change, but "three strings" isn't right, there are only two strings in both examples.

**Comment by jordansissel** [comment]
Ignoring [jenkins failure](http://build-eu-00.elastic.co/job/Logstash_PR/3055/console) because it's metrics-related, not related to this PR.

**Comment by jordansissel** [comment]
LGTM

**Comment by suyograo** [comment]
I really don't know why Jenkins is building PRs now. I removed it a while back.
